flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/TestingLeaderElectionService.java

```diff
@@ -72,31 +85,36 @@ public synchronized boolean hasLeadership(@Nonnull UUID leaderSessionId) {
 		confirmationFuture = new CompletableFuture<>();
 		hasLeadership = true;
 		issuedLeaderSessionId = leaderSessionID;
-		contender.grantLeadership(leaderSessionID);
+
+		if (contender != null) {
+			contender.grantLeadership(leaderSessionID);
+		}
 
 		return confirmationFuture;
 	}
 
 	public synchronized void notLeader() {
 		hasLeadership = false;
-		contender.revokeLeadership();
-	}
 
-	public synchronized void reset() {
-		contender = null;
-		hasLeadership  = false;
+		if (contender != null) {
+			contender.revokeLeadership();
+		}
 	}
 
 	public synchronized String getAddress() {
-		return contender.getAddress();
+		if (contender != null) {
+			return contender.getAddress();
+		} else {
+			throw new IllegalStateException("TestingLeaderElectionService has not been started.");
+		}
 	}
 
 	/**
 	 * Returns <code>true</code> if {@link #start(LeaderContender)} was called,
 	 * <code>false</code> otherwise.
 	 */
-	public synchronized boolean isStarted() {
-		return contender != null;
+	public synchronized CompletableFuture<Void> isStarted() {
```

Comment:
Javadoc is not correct anymore. 
Method name `isStarted` is deceiving.

Reply:
Will update it.